File: TOKI-Learning/pascal/kasur_rusak.pas
function polidrom(a : string):boolean;
begin
	if length(a) = 0 then
		polidrom := true
	else if a[1] = a[length(a)] then
	begin
		delete(a,1,1);
		delete(a,length(a),1);
		polidrom := polidrom(a);
	end
	else
		polidrom := false;
end;
var 
	a : string;
begin
	readln(a);
	if(polidrom(a)) then
		writeln('YA')
	else
		writeln('BUKAN');
end.

File: TOKI-Learning/c++/kasur_rusak.cpp
#include <cstdio>
#include <cstring>

char s[101];

bool polidrom(int awal, int akhir)
{
    if (akhir <= awal)
        return true;
    else if (s[awal] != s[akhir])
        return false;
    else
        return polidrom(awal + 1, akhir - 1);
}

int main()
{
    scanf("%s", s);
    if (polidrom(0, strlen(s) - 1))
        printf("YA\n");
    else
        printf("BUKAN\n");
    return 0;
}
